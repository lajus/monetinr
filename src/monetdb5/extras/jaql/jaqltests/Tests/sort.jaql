# from the docs

nums = [ 2, 1, 3 ];
nums -> sort by [ $ ];
nums -> sort each n by [ n ];
nums -> sort by [ $ desc ];

test = [[2,2,"first"],[1,2,"second"],[2,1,"third"],[1,1,"fourth"]];
test -> sort by [$[0], $[1] desc];
