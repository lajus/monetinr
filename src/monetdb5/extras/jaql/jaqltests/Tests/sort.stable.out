stdout of test 'sort` in directory 'monetdb5/extras/jaql/jaqltests` itself:


# 13:53:50 >  
# 13:53:50 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/net/sofia.ins.cwi.nl/export/scratch1/fabian/tmp/mtest-jacqueline-sofia.ins.cwi.nl/five/dbfarm" "--set" "mapi_open=true" "--set" "mapi_port=32326" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_extras_jaql_jaqltests" "--set" "mal_listing=0"
# 13:53:50 >  

# MonetDB 5 server v11.10.0 "jacqueline-c183b5316fe7+"
# Serving database 'mTests_extras_jaql_jaqltests', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.630 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2013 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://sofia.ins.cwi.nl:32326/
# MonetDB/GIS module loaded
# MonetDB/JAQL module loaded
# MonetDB/SQL module loaded

Ready.

# 13:53:50 >  
# 13:53:50 >  "mclient" "-ljaql" "-ftest" "-Eutf-8" "-i" "-e" "--host=sofia" "--port=32326"
# 13:53:50 >  

#nums -> sort by [ $ ];
% .json # table_name
% json # name
% clob # type
% 11 # length
[ 1, 2, 3 ]
#nums -> sort each n by [ n ];
% .json # table_name
% json # name
% clob # type
% 11 # length
[ 1, 2, 3 ]
#nums -> sort by [ $ desc ];
% .json # table_name
% json # name
% clob # type
% 11 # length
[ 3, 2, 1 ]
#test -> sort by [$[0], $[1] desc];
% .json # table_name
% json # name
% clob # type
% 80 # length
[ [ 1, 2, "second" ], [ 1, 1, "fourth" ], [ 2, 2, "first" ], [ 2, 1, "third" ] ]

# 13:53:50 >  
# 13:53:50 >  "Done."
# 13:53:50 >  

